### PR TITLE
Support `AbortController`

### LINKED ISSUE #78: Support `AbortController`
See https://github.com/sindresorhus/p-queue/issues/23

### PR DESCRIPTION
**Pull request by Richienb**
Fixes #78

### CODE REVIEW

File: test/test.ts [hunk 1]
```diff
@@ -1062,3 +1062,25 @@ test('should verify timeout overrides passed to add', async t => {
...
 	await queue.onIdle();
 });
+
+test('should skip an aborted job', t => {
+	const queue = new PQueue();
+
+	const controller = new AbortController();
+
+	controller.abort();
+	queue.add(() => {
```
Comment (Richienb) [inline test/test.ts:1072]: An aborted job will never ever resolve. Perhaps it should reject instead?

Comment (sindresorhus) [reply test/test.ts:1072]: What do other promise APIs that support AbortController do in this case?

Comment (Richienb) [reply test/test.ts:1072]: `window.fetch` throws `Uncaught (in promise) DOMException: The user aborted a request.`

Comment (sindresorhus) [reply test/test.ts:1072]: Yeah, should abort then. Just use a normal error for now, with a TODO comment to use https://nodejs.org/api/errors.html#abort_err

File: readme.md [hunk 2]
```diff
@@ -131,6 +131,37 @@ Default: `0`
 
 Priority of operation. Operations with greater priority will be scheduled first.
 
+##### signal
+
+Type: `AbortSignal`
+
+AbortSignal for cancellation of the operation. When aborted, it will be removed from the queue. If the operation is already running, the signal will need to be handled by the operation itself.
```
Comment (sindresorhus) [inline readme.md:138]: Linkify `AbortSignal` to the Node.js docs about it (or MDN).

File: source/index.ts [hunk 3]
```diff
@@ -18,41 +18,41 @@ const timeoutError = new TimeoutError();
...
-export default class PQueue<QueueType extends Queue<RunFunction, EnqueueOptionsType> = PriorityQueue, EnqueueOptionsType extends QueueAddOptions = DefaultAddOptions> extends EventEmitter<'active' | 'idle' | 'add' | 'next' | 'completed' | 'error'> {
-	private readonly _carryoverConcurrencyCount: boolean;
+export default class PQueue<QueueType extends Queue<RunFunction, EnqueueOptionsType> = PriorityQueue, EnqueueOptionsType extends QueueAddOptions = QueueAddOptions> extends EventEmitter<'active' | 'idle' | 'add' | 'next' | 'completed' | 'error'> {
+	readonly #carryoverConcurrencyCount: boolean;
 
-	private readonly _isIntervalIgnored: boolean;
+	readonly #isIntervalIgnored: boolean;
 
-	private _intervalCount = 0;
+	#intervalCount = 0;
```
Comment (sindresorhus) [inline source/index.ts:26]: For the future, unrelated changes like this are better as a separate PR.

**Comment by sindresorhus** [comment]
> with a TODO comment to use - https://github.com/sindresorhus/p-queue/pull/161#discussion_r795184998

**Comment by Richienb** [comment]
> > with a TODO comment to use - [#161 (comment)](https://github.com/sindresorhus/p-queue/pull/161#discussion_r795184998)

Already added it: https://github.com/sindresorhus/p-queue/pull/161/files#diff-fcd423a0b6ae0654240ee2c1cb0b023707df57b7d1c7a3834c5bfbef7d901da5R249

**Comment by sindresorhus** [comment]
Great! Thank you :)

**Comment by sindresorhus** [comment]
@Richienb Are you interested in being added as a maintainer here?

**Comment by Richienb** [comment]
@sindresorhus Yeah, sounds fun

**Pull request merged**